player.py

```diff
@@ -0,0 +1,340 @@
+
+        Returns:
+
+        enum:Decision for player's current hand of cards
+
+        """
+
+        # Var to store the player's decision for the current hand of cards 
+        decision = definitions.Actions.HIT
+
+        # Var to store hand total
+        total = self.valid_total(self.hand_total(hand))
+
+        # Var to store numeric value of dealer's upcard 
+        dealer_up_val = self.value_to_int(dealer_upcard)
+
+        # TODO: Rework basic strategy logic.
+
+        # Case for Hard Hands
+        if not self.is_soft(hand):
+            if total in range(5, 8):
+                decision = definitions.Actions.HIT
+            
+            if total == 8:
+                if dealer_up_val in range(5, 7):
+                    decision = definitions.Actions.DOUBLE
+                else:
+                    decision = definitions.Actions.HIT
+            
+            if total == 9:
+                if dealer_up_val in range(2, 7):
+                    decision = definitions.Actions.DOUBLE
+                else:
+                    decision = definitions.Actions.HIT
+            
+            if total == 10:
+                if dealer_up_val in range(2, 10):
+                    decision = definitions.Actions.DOUBLE
+                else:
+                    decision = definitions.Actions.HIT
+            
+            if total == 11:
+                decision = definitions.Actions.DOUBLE
+
+            if total == 12:
+                if dealer_up_val in range(4, 7):
+                    decision = definitions.Actions.STAND
+                else:
+                    decision = definitions.Actions.HIT
+            
+            if total in range(13, 17):
+                if dealer_up_val in range(2, 7):
+                    decision = definitions.Actions.STAND
+                else:
+                    decision = definitions.Actions.HIT
+            
+            if total >= 17:
+                decision = definitions.Actions.STAND
+
+        # Case for Soft Hands
+        else:
+            if total in range(13, 17):
+                if dealer_up_val in range(4, 7):
+                    decision = definitions.Actions.DOUBLE
+                else:
+                    decision = definitions.Actions.HIT
+            
+            if total == 17:
+                if dealer_up_val in range(2, 7):
+                    decision = definitions.Actions.DOUBLE
+                else:
+                    decision = definitions.Actions.HIT
+
+            if total == 18:
+                if dealer_up_val in range(3, 7):
+                    decision = definitions.Actions.DOUBLE
+                elif dealer_up_val in range(9, 11):
+                    decision = definitions.Actions.HIT
+                else:
+                    decision = definitions.Actions.STAND
+            
+            if total == 19:
+                if dealer_up_val == 6:
+                    decision = definitions.Actions.DOUBLE
```

Comment:
I think that the compute_play function in the player class will require a boolean value to keep track if there are split aces. This can be determined in the game class which can call the function with this value. 

It would be easier for the game class if the soft Dh or Ds options on the basic strategy table did not return double in the case of split aces.